src-tauri/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+use headless_chrome::browser::tab;
+use headless_chrome::protocol::cdp::Page;
+use headless_chrome::protocol::cdp::Target::CreateTarget;
+use headless_chrome::{browser, Browser, LaunchOptions};
+use regex::Regex;
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use tokio::sync::Mutex;
+use totp_rs::{Algorithm, Secret, TOTP};
+
+// Login with TOTP
+// Using the chromiumoxide library to handle the login process for the website
+pub async fn request_token(
+    browser: Browser,
+) -> Result<(String, String), Box<dyn std::error::Error>> {
+    const LOGIN_URL: &str =
+        "http://ec2-3-110-151-1.ap-south-1.compute.amazonaws.com/tmp/simulate/login.html";
+    const WAITING_TIME: u64 = 3; //s
+    let new_tab = browser.new_tab()?;
+    let navigated_tab = new_tab.navigate_to(LOGIN_URL)?.wait_until_navigated()?;
+
+    // Wait for the page to load
+    // Step 2: Enter username and password
+    navigated_tab
+        .find_element("input#userid")?
+        .type_into("AB0101")?;
```

Comment:
This is hard-coded and instead should be passed into this function just like the `browser` argument.

Reply:
✅